org_fedora_oscap/service/installation.py

```diff
@@ -25,92 +25,76 @@
 
 from org_fedora_oscap import common, data_fetch, rule_handling, utils
 from org_fedora_oscap.common import _, get_packages_data, set_packages_data
+from org_fedora_oscap.content_handling import ContentCheckError
+from org_fedora_oscap import content_discovery
 
 log = logging.getLogger(__name__)
 
 
 REQUIRED_PACKAGES = ("openscap", "openscap-scanner",)
 
 
-class FetchContentTask(Task):
-    """The installation task for fetching the content."""
+def _handle_error(exception):
+    log.error("Failed to fetch and initialize SCAP content!")
 
-    def __init__(self, policy_data, file_path, content_path):
-        """Create a task."""
-        super().__init__()
-        self._policy_data = policy_data
-        self._file_path = file_path
-        self._content_path = content_path
-
-    @property
-    def name(self):
-        return "Fetch the content"
+    if isinstance(exception, ContentCheckError):
+        msg = _("The integrity check of the security content failed.")
+        terminate(msg)
+    elif (isinstance(exception, common.OSCAPaddonError)
+        or isinstance(exception, data_fetch.DataFetchError)):
+        msg = _("There was an error fetching and loading the security content:\n" +
+                f"{str(exception)}")
+        terminate(msg)
 
-    def run(self):
-        """Run the task."""
-        # Is the content available?
-        if os.path.exists(self._content_path):
-            log.debug("Content is already available. Skip.")
-            return
+    else:
+        msg = _("There was an unexpected problem with the supplied content.")
+        terminate(msg)
 
-        if os.path.exists(self._file_path):
-            log.debug("Content is already available. Skip.")
-            return
 
-        try:
-            data_fetch.fetch_data(
-                self._policy_data.content_url,
-                self._file_path,
-                self._policy_data.certificates
-            )
+def terminate(message):
+    message += "\n" + _("The installation should be aborted.")
+    message += " " + _("Do you wish to continue anyway?")
```

Comment:
Just a note, our handler of the `NonCriticalInstallationError` exception produces a dialog with the "Would you like to ignore this and continue with installation?" question (see https://github.com/rhinstaller/anaconda/pull/3147#issuecomment-773557362). It might be confusing to ask two questions.

Reply:
Is there a difference between RHEL8 Anaconda and the RHEL9 one? I recall that I have seen an error message without the question probably in the RHEL8 GUI, and there were yes/no buttons at the bottom of the message window, which looked weird without the actual question.
I am not 100% sure how to reproduce this, but I can try. Do you have any idea what was it about?

Reply:
In RHEL 8, the addon had to handle a non critical error and show a dialog. In RHEL 9, the addon is not able to handle a non critical error, because the installation tasks are run in a DBus module, so the installer (the process with UI) catches and handles the `NonCriticalInstallationError` error during the installation.

You can try to modify the installation task to make sure that it will raise the error.